packages/core/src/utilities/VoxelManager.ts

```diff
@@ -250,24 +458,141 @@ export default class VoxelManager<T> {
+    const index = Array.isArray(point)
+      ? point[0] + this.width * point[1] + this.frameSize * point[2]
+      : point;
+    if (!this.points) {
+      this.points = new Set<number>();
+    }
+    this.points.add(index);
+  }
+
+  /**
+   * Gets the list of added points as an array of Point3 values
+   */
+  public getPoints(): Point3[] {
+    return this.points
+      ? [...this.points].map((index) => this.toIJK(index))
+      : [];
+  }
+
+  /**
+   * Retrieves the slice data for a given slice view.
+   *
+   * @param sliceViewInfo - An object containing information about the slice view.
+   * @param sliceViewInfo.sliceIndex - The index of the slice.
+   * @param sliceViewInfo.slicePlane - The axis of the slice (0 for YZ plane, 1 for XZ plane, 2 for XY plane).
+   * @returns A typed array containing the pixel data for the specified slice.
+   * @throws Error if an invalid slice axis is provided.
    */
-  public getPixelData: (
-    sliceIndex?: number,
-    pixelData?: PixelDataTypedArray
-  ) => PixelDataTypedArray;
+  public getSliceData = ({
+    sliceIndex,
+    slicePlane,
+  }: {
+    sliceIndex: number;
+    slicePlane: number;
+  }): PixelDataTypedArray => {
+    const [width, height, depth] = this.dimensions;
+    const frameSize = width * height;
+    const startIndex = sliceIndex * frameSize;
+
+    let sliceSize: number;
+    const SliceDataConstructor = this.getConstructor();
+
+    function isValidConstructor(
+      ctor: any
+    ): ctor is new (length: number) => PixelDataTypedArray {
+      return typeof ctor === 'function';
+    }
+
+    if (!isValidConstructor(SliceDataConstructor)) {
+      // Return an empty typed array instead of an empty regular array
+      return new Uint8Array(0) as PixelDataTypedArray;
+    }
+
+    // Todo: optimize it when we have scalar data
+    let sliceData: PixelDataTypedArray;
+    switch (slicePlane) {
+      case 0: // YZ plane
+        sliceSize = height * depth;
+        sliceData = new SliceDataConstructor(sliceSize);
+        for (let i = 0; i < height; i++) {
+          for (let j = 0; j < depth; j++) {
+            const index = sliceIndex + i * width + j * frameSize;
+            this.setSliceDataValue(sliceData, i * depth + j, this._get(index));
+          }
+        }
+        break;
+      case 1: // XZ plane
+        sliceSize = width * depth;
+        sliceData = new SliceDataConstructor(sliceSize);
+        for (let i = 0; i < width; i++) {
+          for (let j = 0; j < depth; j++) {
+            const index = i + sliceIndex * width + j * frameSize;
+            this.setSliceDataValue(sliceData, i + j * width, this._get(index));
+          }
+        }
+        break;
+      case 2: // XY plane
+        sliceSize = width * height;
+        sliceData = new SliceDataConstructor(sliceSize);
+        for (let i = 0; i < sliceSize; i++) {
+          this.setSliceDataValue(sliceData, i, this._get(startIndex + i));
+        }
+        break;
+      default:
```

Comment:
If the slice plane was a normal, then it would be possible to implement this - maybe add a comment like
// Oblique plane - todo - implement as ortho normal vector.